### PR TITLE
Plume front arm support

### PR DESCRIPTION
**Pull request by mcrosson**
This PR addresses #394 and #400

Changes

- Update Dockerfile to add necessary lld dependency on arm platforms via a helper script
- Run the build/install of plume-front with the necessary linker=lld option on arm platforms via a helper script
- Add rust-toolchain to Docker container along with Cargo files to ensure the right rust toolchain is selected for later steps

Builds Tested On
- x86-64
- arm64 (aarch64)
- arm32 (armv7l)

Noteworthy items

- The rust wasm packages do NOT include rustc-lld for arm
- The lld binaries published by llvm.org will NOT build plume-front with linker=lld rust args
- lld built from sources (what this patch does) on arm DOES build plume-front with linker=lld rust args

### CODE REVIEW

**Comment by fdb-hiroshima** [comment]
I'd prefer `plume-front.sh` and `wasm-deps.sh` to be put in the `script` folder, else it seems all good

File: Dockerfile [hunk 1]
```diff
@@ -10,12 +10,15 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     make \
     openssl \
     libssl-dev
+WORKDIR /scratch
+COPY script/wasm-deps.sh .
+RUN chmod a+x ./wasm-deps.sh && ./wasm-deps.sh
 WORKDIR /app
 COPY Cargo.toml Cargo.lock ./
```
[the remark this answers is not in the document]
Comment (mcrosson) [reply Dockerfile]: I have #400 open for that ; I'll update per your changes above and include #400 in the main PR comments too.

[the remark this answers is not in the document]
Comment (mcrosson) [reply Dockerfile]: No worries, this trick solved something that was bugging me and I'm happy to take care of two tickets with a single PR.

**Comment by fdb-hiroshima** [comment]
Fix #400 
Fix #394

**Comment by vielmetti** [comment]
> The lld binaries published by llvm.org will NOT build plume-front with linker=lld rust args

@mcrosson do you recall what's missing with the llvm distributed lld binaries?  I wonder if that's something that might be fixed upstream.

**Comment by mcrosson** [comment]
@vielmetti IIRC the Rust packaging does not include arm32 or 64 as tier 1 for lld-wasm. I think it's on the WASM side? My memory is mildly hazy, there were a lot of tickets that I read and I'm still not familiar with the rust ecosystem.

**Pull request merged**